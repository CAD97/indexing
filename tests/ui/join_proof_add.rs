use indexing::{scope, Range, proof::NonEmpty};

fn main() {
    let arr1 = [1, 2, 3, 4, 5];

    scope(&arr1[..], |v| {
        let r = v.range();
        if let Some(r) = r.nonempty() {
            let r: Range<'_, _, NonEmpty> = r;
            let (front, back) = r.frontiers();

            r.start();
            front.join(r).unwrap().start();
            r.join(back).unwrap().start();
            front.join_cover(r).start();
            r.join_cover(back).start();
            r.join_cover(r).start();

            // bluss uses `.last()`, which we don't have
            // instead we use `.observe_proof()`
            r.observe_proof();
            front.join(r).unwrap().observe_proof();
            r.join(back).unwrap().observe_proof();
            front.join_cover(r).observe_proof();
            r.join_cover(back).observe_proof();
            r.join_cover(r).observe_proof();

            front.join_cover(back).start();
            front.join_cover(back).observe_proof();
            //~^ ERROR no method named
        }
    });
}
