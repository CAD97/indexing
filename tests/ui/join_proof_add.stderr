error[E0599]: no method named `observe_proof` found for type `indexing::index::range::Range<'_>` in the current scope
  --> $DIR/join_proof_add.rs:28:36
   |
28 |             front.join_cover(back).observe_proof();
   |                                    ^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0599`.
